index.ts

```diff
@@ -1,82 +1,173 @@
+import { init, Ditto, Document, QueryResultItem } from "@dittolive/ditto";
+import * as readline from "readline/promises";
+import { stdin as input, stdout as output } from "node:process";
 
-import { init, Ditto, Document } from '@dittolive/ditto'
-import * as readline from 'readline/promises'
-import { stdin as input, stdout as output } from 'node:process';
+let ditto;
+let tasks: Document[] = [];
+// Attachments dont seem to show up as part of a document. Using Query item instead.
+let attachments: QueryResultItem[] = [];
+let queryResult;
 
-let ditto
-let subscription
-let liveQuery
-let tasks: Document[] = []
-
-async function main () {
-  await init()
+async function main() {
+  await init();
 
   ditto = new Ditto({
-    type: 'onlinePlayground',
-    appID: 'YOUR_APP_ID',
-    token: 'YOUR_TOKEN'
-  })
-  ditto.startSync()
-
-  subscription = ditto.store.collection("tasks").find("isDeleted == false").subscribe()
-  liveQuery = ditto.store.collection("tasks").find("isDeleted == false").observeLocal((docs, event) => {
-    tasks = docs
-  })
-  let isAskedToExit = false
-  
+    type: "onlinePlayground",
+    appID: "REPLACE_WITH_YOUR_APP_ID",
+    token: "REPLACE_WITH_YOUR_TOKEN",
+  });
+  ditto.startSync();
+
+  ditto.sync.registerSubscription(
+    `SELECT * FROM tasks WHERE isDeleted = false`,
+  );
+
+  ditto.sync.registerSubscription(
+    `SELECT * FROM COLLECTION attachments (my_attachment ATTACHMENT) WHERE isDeleted = false`,
+  );
+
+  ditto.store.registerObserver(
+    `SELECT * FROM tasks WHERE isDeleted = false AND isCompleted = false`,
+    (result) => {
+      tasks = result.items.map((doc) => {
+        return doc.value;
+      });
+    },
+  );
+
+  ditto.store.registerObserver(
+    "SELECT * FROM COLLECTION attachments (my_attachment ATTACHMENT) WHERE isDeleted = false AND isCompleted = false",
+    (result) => {
+      attachments = result.items;
+      // Attachments dont seem to show up as part of a document
+      //   .map((doc) => {
+      //   return doc.value;
+      // });
```

Comment:
This query will return attachment tokens, which can then be used to fetch attachments with `const attachment = await ditto.store.fetchAttachment(result.items[i].value.my_attachment)`. 

Does this query really not return values?

Reply:
Ah, sorry, I see it's there in another part of the script, was just reacting to the commented out bit.

Reply:
Assuming we do want to update the template app, @ErikEverson - can you remove the commented out block?

Reply:
Yeah I will remove that. We will probably have to wait until 4.6 is released first as the code will not compile until 4.6 can be used.

Reply:
Code has been cleaned up. Should be ready for merged once 4.6 is done.